experimental/ComplexReflectionGroups/src/complex_reflection_group.jl

```diff
@@ -0,0 +1,1729 @@
+# This file implements explicit models of complex reflection groups.
+#
+# References:
+#
+# * Lehrer, G. I., & Taylor, D. E. (2009). Unitary reflection groups (Vol. 20, p. viii). Cambridge University Press, Cambridge.
+# 
+# * Thiel, U. (2014). On restricted rational Cherednik algebras. TU Kaiserslautern.
+#
+# * Marin, I., & Michel, J. (2010). Automorphisms of complex reflection groups. Represent. Theory, 14, 747–788.
+#
+# Ulrich Thiel, 2023 
+
+export complex_reflection_group
+export is_complex_reflection_group
+export complex_reflection_group_type
+export complex_reflection_group_model
+export complex_reflection_group_dual
+
+function complex_reflection_group(G::ComplexReflectionGroupType; model=:CHEVIE)
+    
+    # this will be the list of matrix groups corresponding to the components of G
+    component_groups = MatrixGroup[]
+
+    # list of models
+    modellist = []
+
+    for C in components(G)
+
+        t = C.type[1]
+        gens = nothing
+        Cmodel = nothing
+
+         # Set default model (CHEVIE)
+         if model === nothing
+            Cmodel = :CHEVIE
+        else
+            Cmodel = model
+        end
+
+        # we now create the list "gens" of matrix generators for C in the selected model
+        if isa(t, Int)
+
+            # Exceptional groups
+
+            # G4
+            if t == 4
+
+                if Cmodel == :LT
+                    # See Lehrer & Taylor (2009), page 85-86
+                    R,x = polynomial_ring(QQ)
+                    K,i = number_field(x^2+1, "i")
+                    R,x = polynomial_ring(K)
+                    K,ω = number_field(x^2+x+1, "ω")
+                    i = K(i)
+                    matspace = matrix_space(K, 2, 2)
+                    gens = elem_type(matspace)[]
+
+                    r1 = ω//2 * matspace(K[-1-i 1-i ; -1-i -1+i])
+                    r1prime = ω//2 * matspace(K[-1-i -1+i ; 1+i -1+i])
+
+                    push!(gens, r1)
+                    push!(gens, r1prime)
```

Comment:
Alternatively, do this:
```suggestion
                    gens = [
                       ω//2 * matrix(K, [-1-i 1-i ; -1-i -1+i]),  # r1
                       ω//2 * matrix(K, [-1-i -1+i ; 1+i -1+i])   # r1'
                    ]
```

Reply:
Cheers, I'll take care of this in my next coding session.

Reply:
Note to self: transpose matrices for the unitary models by LT because of x*A action in OSCAR.

Reply:
I'll leave it as is since this is somehow too much changing with no real effect.